### ISSUE TITLE
Implement \cs_use:c

### ISSUE THREAD

**Issue by Vidabe** [latex3/latex3#516]
I think that it might come in handy to have a `\cs_use:c` command (which would
work equivalent to `\tl_use:c` and friends) as the disadvantage of `\use:c` is that it
doesn't check if the command exists at all. 

The main reason for my request is that I alway make the mistake of thinking "`\tl_use:c` checks if
the variable exists so `\use:c` does too" which always leads to a bit of debugging as I 
(mistakenly) rely on the (non-existent) error-checking.

**Comment by wspr** [comment]
Why is that an issue? Isn’t it orthogonal to the case at hand? I’m not sure, incidentally, why we couldn't have a \cs_new:N to precede \cs_set… 

To me the request is simply that \cs_use:c would be a useful command to have (presumably to catch typos or incorrect code logic).

**Comment by car222222** [comment]
Yes, Will, I see the distinction.  But I still think that \use:c must be renamed as \cs_use:c.

So the error message version would need a new name, maybe \cs_use_or_error:c .

But, as Joseph wrote, there is still no valid comparison with \tl . . .  commands.

I agree that if \use:c is weeded out, all we need to do is to explain that \use . . .  always refers to doing things with token lists (I.e. the tl_ is omitted from their names).

**Comment by Vidabe** [comment]
Apologies for the late reply (and for possible broken english; I am tired). But yes, as Will said, the idea is to catch typos and incorrect logic (but mostly typos) and having a `\use:c` command which enforces that the csname exists. It is for cases where I *know* that the csname *has to* exist. I wonder if it would also help to write a bit of cleaner code, as it would theoretically "enforce" people to define commands via `\cs_new` first before they use them in `\cs_set`. (not quite sure if I understand the problem) (of course `\use:c` would still exist).

I also don't think that the existence of `\cs_if_exist_use:c`  is a good reason to not implement it.
Using the same argument I could propose that `\int_incr:N` should be deleted as one could simply write `\int_add:Nn \cs { 1 }` (same for `int_decr:N `), or that there is no need for `\str_clear_new:N` as it is just a shortcut for `\str_if_exist:NTF` in combination with `\str_new:N` and `\str_clear:N`. Or -- a more general example -- there is no need for `C` (and others) to allow the syntax `++int` if one could (and can) simply write `int = int + 1`. But being able to write`++int` is just much more handy then the alternative (at least as someone who writes in Fortran I imagine it to be more handy).

Renaming `\use:c` to `\cs_use:c` will probably create more confusion, as the parallels to the likes of `\tl_clear:c` would be (wrongly) more obvious. Maybe something like `\use_as_cs:c`? It would show that it does something with tokens (like the other `\use` commands) and that it creates a command sequence. And the command with error checking would be `\use_as_cs_or_error:c `? Bit clunky, but it would work.

**Comment by wspr** [comment]
FWIW I've started implementing some of the changes Bruno outlined above. I'll make it public for discussion once I've ironed out some wrinkles and checked for more consistency across variables.

**Comment by wspr** [comment]
I just realised I am now years behind trying to complete this change. I got stalled when my pull request no longer cleanly merged :) I will try to get back to this as a priority...
